plugins/woocommerce/includes/widgets/class-wc-widget-cart.php

```diff
@@ -70,8 +106,17 @@ public function widget( $args, $instance ) {
 			echo '<div class="hide_cart_widget_if_empty">';
 		}
 
-		// Insert cart widget placeholder - code in woocommerce.js will update this on page load.
-		echo '<div class="widget_shopping_cart_content"></div>';
+		global $post;
+		$is_cart_related_block = $this->array_contains( $post->post_content, self::get_blocks() );
+
+		if ( $is_cart_related_block || is_shop() || is_product() || is_product_category() ) {
+			wp_enqueue_script( 'wc-cart-fragments' );
+			// Insert cart widget placeholder - code in woocommerce.js will update this on page load.
+			echo '<div class="widget_shopping_cart_content"></div>';
+		} else {
+			// Insert cart widget with the Mini-cart content.
+			echo '<div class="widget_shopping_cart_content">' . woocommerce_mini_cart() . '</div>';
```

Comment:
woocommerce_mini_cart should be escaped with `esc_html`

Reply:
Done!